vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -847,6 +853,59 @@ impl<T: DeviceBacking> AsyncRun<WorkerState> for DriverWorkerTask<T> {
 }
 
 impl<T: DeviceBacking> DriverWorkerTask<T> {
+    /// Select the optimal CPU for an interrupt vector to achieve better distribution.
+    /// This implements a stride-based algorithm to spread interrupt vectors across CPUs
+    /// when there are significantly more CPUs than interrupt vectors.
+    ///
+    /// To avoid overlap between multiple NVMe driver instances, this method incorporates
+    /// a device-specific offset based on the device ID hash.
+    fn select_cpu_for_interrupt(&self, iv: u16, requesting_cpu: u32) -> u32 {
+        // Total number of CPUs available
+        let cpu_count = self.io_issuers.per_cpu.len() as u32;
+        let max_interrupt_count = self.device.max_interrupt_count().max(1);
+
+        // Only apply stride-based distribution if we have significantly more CPUs than interrupt vectors
+        // and we have more than 4 interrupt vectors (to avoid breaking existing tests)
+        if cpu_count > max_interrupt_count * 2 && max_interrupt_count > 4 {
+            // Calculate stride to distribute interrupt vectors across CPUs
+            let stride = cpu_count / max_interrupt_count;
+            let stride = stride.max(1); // Ensure stride is at least 1
```

Comment:
You have already checked that `cpu_count` is divisible by `max_interrupt_count`, so `stride` must always be greater than or equal to 1.